sympy/polys/rings.py

```diff
@@ -914,47 +838,40 @@ def is_nonpositive(self):
         return self.ring.domain.is_nonpositive(self.LC)
 
     @property
-    def is_zero(f):
-        return not f
+    def is_zero(self):
+        return self._is_zero()
 
     @property
-    def is_one(f):
-        return f == f.ring.one
+    def is_one(self):
+        return self == self.ring.one
 
     @property
-    def is_monic(f):
-        return f.ring.domain.is_one(f.LC)
+    def is_monic(self):
+        return self.ring.domain.is_one(self.LC)
 
     @property
-    def is_primitive(f):
-        return f.ring.domain.is_one(f.content())
+    def is_linear(self):
+        return all(sum(monom) <= 1 for monom in self.itermonoms())
 
     @property
-    def is_linear(f):
-        return all(sum(monom) <= 1 for monom in f.itermonoms())
+    def is_primitive(self):
+        return self.ring.domain.is_one(self.content())
 
     @property
-    def is_quadratic(f):
-        return all(sum(monom) <= 2 for monom in f.itermonoms())
+    def is_quadratic(self):
+        return all(sum(monom) <= 2 for monom in self.itermonoms())
 
     @property
-    def is_squarefree(f):
-        if not f.ring.ngens:
-            return True
-        return f.ring.dmp_sqf_p(f)
+    def is_squarefree(self):
+        return self._is_squarefree()
 
     @property
-    def is_irreducible(f):
-        if not f.ring.ngens:
-            return True
-        return f.ring.dmp_irreducible_p(f)
+    def is_irreducible(self):
+        return self._is_irreducible()
```

Comment:
What is the purpose of separating `is_irreducible` from `_is_irreducible` here?

If one just calls the other then I don't see what the idea is.